spock-core/src/main/java/org/spockframework/runtime/SpockRuntime.java

```diff
@@ -34,21 +34,44 @@ public static void verifyCondition(@Nullable ValueRecorder recorder,
       @Nullable String text, int line, int column, @Nullable Object message, @Nullable Object condition) {
     if (!GroovyRuntimeUtil.isTruthy(condition)) {
       throw new ConditionNotSatisfiedError(
-          new Condition(getValues(recorder), text, TextPosition.create(line, column), messageToString(message)));
+          new Condition(getValues(recorder), text, TextPosition.create(line, column), messageToString(message), null, null));
     }
   }
 
+  public static final String CONDITION_FAILED_WITH_EXCEPTION = "conditionFailedWithException";
+
+  public static void conditionFailedWithException(@Nullable ValueRecorder recorder, @Nullable String text, int line, int column, @Nullable Object message, Throwable throwable){
+    if (throwable instanceof SpockAssertionError){
+      throw (SpockAssertionError) throwable; // this is our exception - it already has good message
+    }
+    if (throwable instanceof SpockException){
+      throw (SpockException) throwable; // this is our exception - it already has good message
+    }
+    throw new ConditionNotSatisfiedError(
```

Comment:
I'm not an expert in a topic of OOM, so I may be wrong, however it some cases it could be not necessary to shutdown the whole JVM in case of OOM (like a try to allocate a large memory block in a particular thread). In that case a try to render it as a "normal" error could succeed. Even if not another OOM error will be thrown (maybe with less interesting stack trace), but it is just a test - it should be possible to find out why it happened anyway. There is a risk to keep tests execution in an undefined state, but I think we talk here about quite rare situations and the more programmatic approach would to keep this useful PR as is to make it easier/faster to merge and - if needed - come back to that topic after the first related bug report. What do you think @leonard84?